src/python/WMCore/Services/Rucio/Rucio.py

```diff
@@ -607,12 +607,13 @@ def listDataRules(self, name, **kwargs):
         """
         kwargs["name"] = name
         kwargs.setdefault("scope", "cms")
-        res = []
         try:
             res = self.cli.list_replication_rules(kwargs)
-        except Exception as ex:
-            self.logger.error("Exception listing rules for data: %s and kwargs: %s. Error: %s",
-                              name, kwargs, str(ex))
+        except Exception as exc:
+            msg = "Exception listing rules for data: {} and kwargs: {}. Error: {}".format(name,
+                                                                                          kwargs,
+                                                                                          str(exc))
+            raise WMRucioException(msg)
```

Comment:
I am leaving the comment here, which is related to an old piece of code referenced here, but I cannot make the comment on the exact place the definition happens. Anyway, since we are about to raise those `WMRucioException' exceptions here and there, I would suggest to have a little bit more "by the standard' definition on line 23 [1]. The bare minimum I would say should be to have at least an `__init__` method, instead of just `pass`. Something like:
```
class WMRucioException(WMException):
    """
    _WMRucioException_
    Generic WMCore exception for Rucio
    """
    def __init__(self, message):
        self.myMessage = "WMRucioException: %s" % message
        super(WMRucioException, self).__init__(self.myMessage)
```

[1]
https://github.com/dmwm/WMCore/blob/ea35a220e9cd1277f52afb2db730e6a155ba7323/src/python/WMCore/Services/Rucio/Rucio.py#L23

Reply:
My understanding is, since we do not define a constructor `__init__` method here, when we create an instance of WMRucioException, it actually goes to the parent class constructor class (thus executing the WMException `__init__` method). Which is exactly what you're asking for here.

Reply:
Yes, your understanding is correct and yes what I was talking about is 'almost' the same. There is one slight difference though. Notice the line:
`        self.myMessage = "WMRucioException: %s" % message` 
Which basically adds the child exception signature prefixing the message passed to the parent exception class. Which is kind of a simple way to reveal the exception hierarchy. But anyway it is not important, I already approved your change so no need to waste time on that.

Reply:
Oh, I see. Yes, it isn't clear to me how useful it would be to have the exception type as part of the error message (especially for the "end user"). I think the most important is to have a clear error message (and be able to distinguish them, in case there are possibly many of them). That's a valid comment though, thanks.